pyproject.toml

```
@@ -72,12 +72,13 @@ arelleGUI = "arelle.CntlrWinMain:main"
 [tool.setuptools]
 platforms = ["any"]
 
-[tool.setuptools.dynamic]
-version = {attr = "arelle.Version.__version__"}
-
 [tool.setuptools.packages.find]
 namespaces = false
 
+[tool.setuptools_scm]
+tag_regex = "^(?:[\\w-]+-?)?(?P<version>[vV]?\\d+(?:\\.\\d+){0,2}[^\\+]*)(?:\\+.*)?$"
```

The only difference between this and the default regex is making a `-` optional between a tag version prefix and the version itself (`edgr19.2.1` vs `edgr-19.2.1`). We can drop this in the future if we're able to delete (or rename) the old edgar tags that have that format.
```suggestion
tag_regex = "^(?:[\\w-]+-)?(?P<version>[vV]?\\d+(?:\\.\\d+){0,2}[^\\+]*)(?:\\+.*)?$"
```